contents/swift/transition-to-swift.md

```diff
@@ -0,0 +1,406 @@
+```
+
+#### Implicit returns from single-expression closures:
+Omit return
+```
+reversedNames = names.sorted(by: { s1, s2 in s1 > s2 } )
+```
+
+#### Shorthand argument names:
+Omit argument list, use `$0` and `$1` refer to the closure’s first and second String arguments.
+```
+reversedNames = names.sorted(by: { $0 > $1 } )
+```
+
+#### Trailing closure syntax:
+Trailing closures are most useful when the closure is sufficiently long that it is not possible to write it inline on a single line.  
+```
+reversedNames = names.sorted() { $0 > $1 }
+```
+If the closure expression is the only argument, you can also discard the `()` following the function’s name
+```
+reversedNames = names.sorted { $0 > $1 }
+```
+Isn't it much neater now?
+
+### Escaping and Nonescaping Closures in Swift 3:
+A non-escaping closure means that the closure that is passed as an argument to a function will be invoked before the function returns.
+
+On the other hand, if a closure is passed as an argument to a function and it is invoked after the function returns, the closure is escaping.
+
+By default, closures are non-escaping in Swift 3. If the compiler knows that a closure is non-escaping, it can take care of a many of the nitty-gritty details of memory management.
+
+More in-depth explanations and examples:
+https://swiftunboxed.com/lang/closures-escaping-noescape-swift3/
+https://cocoacasts.com/what-do-escaping-and-noescaping-mean-in-swift-3/
+
+### Autoclosures
+You are unlikely to need to use autoclosures, but if you're curious, or if you ever see the term `@autoclosure`:
+An autoclosure is a closure that is automatically created to wrap an expression that’s being passed as an argument to a function.
+
+Examples of autoclosures:
+https://cocoacasts.com/how-to-use-autoclosures-and-autoclosure-in-swift-3/
+
+
+# Others
+Once you start to get used to Swift and expand your project, you might come across new terms.
+
+- Difference between Cocoa and Cocoa Touch
+
+  Quoted from a stackoverflow answer comment:
+  "Cocoa is for Mac development; Cocoa Touch is for iOS development. If something is only in Cocoa, you can't use it on iOS, and if something is only in Cocoa Touch, you can't use it on Mac OS X."
+
+  http://stackoverflow.com/questions/2297841/cocoa-versus-cocoa-touch-what-is-the-difference
+  http://www.whoishostingthis.com/resources/cocoa/
+
+
+- CocoaPods
+  If you are the kind of person that prefers to do everything yourself, this might be irrelevant to you. Otherwise, CocoaPods is the package manager for Swift. You are likely to use this if you depend on libraries a lot.
+
+  - Getting Started and more information<br>
+  https://blog.versioneye.com/2014/01/15/which-programming-language-has-the-best-package-manager/<br>
+  https://guides.cocoapods.org/using/getting-started.html
+  https://dzone.com/articles/concise-introduction-cocoapods
+
+  - How to use<br>
+  https://guides.cocoapods.org/using/using-cocoapods.html
```

Comment:
1. Decide if this article is meant for programmers new to Swift but know ObjC or for those totally new to iOS programming. The content can change accordingly.
2. Try to include less nitty-gritty details of the languages as the reader is unlikely to use this doc to learn Swift when there are many other tutorials out there. There is no need to provide lot of motivation for learning Swift either, because it's pretty much the preferred way to write iOS apps anyway. So the article can focus on guiding the reader to the right resources and providing a 'learning path' instead.